File: frontend/olympic_it_project/olympic_it_project/lib/core/page_response.dart
class PageResponse<T> {
  final List<T> data;
  final int totalPages;
  final int totalElements;
  final int page;
  final int size;

  PageResponse({
    required this.data,
    required this.totalPages,
    required this.totalElements,
    required this.page,
    required this.size,
  });

  factory PageResponse.fromJson(
    Map<String, dynamic> json,
    T Function(dynamic) fromJsonT,
  ) {
    return PageResponse(
      data: (json['data'] as List)
          .map((e) => fromJsonT(e))
          .toList(),
      totalPages: json['totalPages'] ?? 0,
      totalElements: json['totalElements'] ?? 0,
      page: json['page'] ?? 0,
      size: json['size'] ?? 0,
    );
  }
}

File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/dto/exam/PageResponse.java
package org.example.olympic_ot_project.dto.exam;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class PageResponse<T> {
    private List<T> data;
    private long totalElements;
    private int totalPages;
    private int page;
    private int size;
}